AppLovin MAX Demo App - ObjC/AppLovin MAX Demo App - ObjC/AppLovin/MRECs/Table View/ALDemoMRecTableViewController.m

```diff
@@ -0,0 +1,126 @@
+//
+//  ALDemoMRecTableViewController.m
+//  AppLovin MAX Demo App - ObjC
+//
+//  Created by Alan Cao on 6/30/22.
+//  Copyright © 2022 AppLovin Corporation. All rights reserved.
+//
+
+#import "ALDemoMRecTableViewController.h"
+#import "ALDemoMRecTableViewCell.h"
+#import <AppLovinSDK/AppLovinSDK.h>
+
+@interface ALDemoMRecTableViewController () <UITableViewDelegate, UITableViewDataSource, MAAdViewAdDelegate>
+
+@property (weak, nonatomic) IBOutlet UITableView *tableView;
+
+@property (nonatomic) NSMutableArray *adViewQueue;
+@property (nonatomic) NSArray *sampleData;
+
+@end
+
+@implementation ALDemoMRecTableViewController
+
+- (void)viewDidLoad
+{
+    [super viewDidLoad];
+    
+    self.adViewQueue = [NSMutableArray array];
+    self.sampleData = @[@"A", @"B", @"C", @"D", @"E", @"F", @"G", @"H", @"I", @"J", @"K", @"L"];
+    
+    // Configure table view
+    self.tableView.al_delegate = self;
+    self.tableView.al_dataSource = self;
+    self.tableView.estimatedRowHeight = 250;
+    self.tableView.rowHeight = 250;
+    
+    [self configureAdViews: 5];
+}
+
+- (void)configureAdViews:(NSInteger)count
+{
+    for (int i = 0; i < count; i++)
+    {
+        MAAdView *adView = [[MAAdView alloc] initWithAdUnitIdentifier: @"YOUR_AD_UNIT_ID" adFormat: MAAdFormat.mrec];
+        adView.delegate = self;
+        
+        // Set this extra parameter to work around SDK bug that ignores calls to stopAutoRefresh()
+        [adView setExtraParameterForKey: @"allow_pause_auto_refresh_immediately" value: @"true"];
+        [adView stopAutoRefresh];
+        
+        // Load the ad
+        [adView loadAd];
+        [self.adViewQueue addObject: adView];
+    }
+}
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView
+{
+    return self.sampleData.count;
+}
+
+- (NSInteger)tableView:(nonnull UITableView *)tableView numberOfRowsInSection:(NSInteger)section
+{
+    return 1;
+}
+
+- (nonnull UITableViewCell *)tableView:(nonnull UITableView *)tableView cellForRowAtIndexPath:(nonnull NSIndexPath *)indexPath
+{
+    ALDemoMRecTableViewCell *cell = [tableView dequeueReusableCellWithIdentifier: @"ALDemoMRecTableViewCell" forIndexPath: indexPath];
```

Comment:
We should be using two different cell types for the ad view cell and the regular data text cell, to reflect a real-world use case. This should involve creating two cell prototypes in the storyboard with their own reuse identifiers. Fortunately the default `UITableViewCell` already has a default `textLabel` that we can utilize for the regular data and then just have `ALDemoMRecTableViewCell` dedicated to only displaying ad views.